addons/mail/static/src/js/thread_field.js

```diff
@@ -115,7 +118,10 @@ var ThreadField = AbstractField.extend({
                 self._threadWidget.render(self._documentThread, {
                     displayLoadMore: self._documentThread.getMessages().length < self._documentThread.getMessageIDs().length,
                 });
-                return self._documentThread.markAsRead();
+                if (self._markAsReadOnNextRender) {
+                    self._documentThread.markAsRead();
```

Comment:
you lose the return value. Is it on purpose?

Reply:
@alexkuhn

Reply:
Not on purpose. I have put it back.